maintenance/wikia/importMessagingWiki.php

```diff
@@ -0,0 +1,179 @@
+<?php
+require_once __DIR__ . '/../Maintenance.php';
+
+class ImportMessagingWiki extends Maintenance {
+
+	/** @var string MESSAGING_WIKI_DB messaging wiki database name */
+	const MESSAGING_WIKI_DB = 'messaging';
+
+	/**
+	 * @var array Array of messages from messaging wiki (same format as in i18n php files
+	 */
+	protected $messages = [];
+
+	/**
+	 * @var bool Whether to modify source files in app repo or write to tmp
+	 */
+	protected $isDryRun = true;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->addOption( 'dry-run', 'Don\'t make any changes to i18n files - print output in /tmp/ folder.' );
+		$this->addOption( 'message', 'Message name to process', false, true /* $withArg */ );
+	}
+
+	/**
+	 * Main entry point
+	 */
+	public function execute() {
+		$this->isDryRun = $this->getOption( 'dry-run', false );
+		$hasData = $this->getMessagingWikiData();
+
+		if ( $hasData ) {
+			$this->processExtensions();
+		}
+	}
+
+	/**
+	 * Load customized messages from messaging wiki
+	 * @return bool if we found any data
+	 */
+	protected function getMessagingWikiData() {
+		global $wgContLang;
+
+		$i = 0;
+
+		$db = wfGetDB( DB_SLAVE, [], static::MESSAGING_WIKI_DB );
+		$sql = ( new WikiaSQL() )
+			->SELECT( 'page_title', 'old_text', 'old_flags' )
+			->FROM( 'page, revision, text' )
+			->WHERE( 'page_namespace' )
+			->EQUAL_TO( NS_MEDIAWIKI )
+			->AND_( 'page_is_redirect' )
+			->EQUAL_TO( 0 )
+			->AND_( 'page_latest' )
+			->EQUAL_TO_FIELD( 'rev_id' )
+			->AND_( 'rev_text_id' )
+			->EQUAL_TO_FIELD( 'old_id' );
+
+		# handle --message argument
+		$msg = $this->getOption('message');
+		if ($msg) {
+			# AND page_title LIKE 'Autocreatewiki-welcomebody/%'
+			$sql->AND_('page_title')->LIKE( ucfirst( $msg ) . '/%' );
+		}
+
+		$res = $sql->runLoop( $db, function ( &$res, $row ) use ( &$i, $wgContLang ) {
+			$text = Revision::getRevisionText( $row );
+			$lckey = $wgContLang->lcfirst( $row->page_title );
+			if ( strpos( $lckey, '/' ) ) {
+				$t = explode( '/', $lckey );
+				$key = $t[0];
+				$lang = $t[1];
+			} else {
+				$key = $lckey;
+				$lang = 'en';
+			}
+
+			$res['messages'][$lang] = $res['messages'][$lang] ?? [];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "messages" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)